docs/secrets.md

```diff
@@ -253,4 +227,101 @@ azure_secret = context.get_secret("MY_AZURE_SECRET")
+
+##### Inline
+The inline secrets provider is a very basic framework that should mostly be used for testing and demos. The secrets 
+passed by this framework are exposed in the source code creating the MLRun function, as well as in the function spec, and
+in the generated pod specs. To add inline secrets to a job, perform the following:
+
+```python
+task.with_secrets("inline", {"MY_SECRET": "12345"})
+```
+
+As can be seen, even the client code exposes the secret value. If this is used to pass secrets to a job running in a kubernetes 
+pod the secret is also visible in the pod spec. This means that any user that can run `kubectl` and is permitted 
+to view pod specs can also see the secret keys and their values.
+
+##### Environment
+Environment variables are similar to the `inline` secrets, but their client-side value is not specified directly in
+code but rather is extracted from a client-side environment variable. For example, if running MLRun on a Jupyter 
+notebook and there are environment variables named `MY_SECRET` and `ANOTHER_SECRET` on Jupyter, the following code  
+passes those secrets to the executed runtime:
+
+```python
+task.with_secrets("env", "MY_SECRET, ANOTHER_SECRET")
+```
+
+When generating the runtime execution environment (for example, pod for the `job` runtime), MLRun retrieves the
+value of the environment variable and places it in the pod spec. This means that a user with `kubectl` capabilities who
+can see pod specs can still see the secret values passed in this manner.
+
+##### File
+The file provider is used to pass secret values that are stored in a local file. The file needs to be made of 
+lines, each containing a secret and its value separated by `=`. For example:
+
+```shell
+# secrets.txt
+SECRET1=123456
+SECRET2=abcdef
+```
+
+Use the following command to add these secrets:
+
+```python
+task.with_secrets("file", "/path/to/file/secrets.txt")
+```
+
+</details>
+
+## Externally managed secrets
+MLRun provides facilities to map k8s secrets that were created externally to jobs that are executed. To enable that,
+the spec of the runtime that is created should be modified by mounting secrets to it - either as files or as 
+environment variables containing specific keys from the secret.
+
+### Mapping secrets to environment
+Let's assume a k8s secret called `my-secret` was created in the same k8s namespace where MLRun is running, with two
+keys in it - `secret1` and `secret2`. The following example will add these two secret keys as environment variables
+to an MLRun job:
+
+```python
+function = mlrun.code_to_function(
+    name="secret_func",
+    handler="test_function",
+    ...
+)
+
+function.set_env_from_secret("SECRET_ENV_VAR_1", secret="my-secret", secret_key="secret1")
+function.set_env_from_secret("SECRET_ENV_VAR_2", secret="my-secret", secret_key="secret2")
+```
+
+This will only take effect for functions executed remotely, as the secret value will be injected to the function pod.
```

Comment:
present tense:
This only takes effect for functions executed remotely, as the secret value is injected to the function pod.
But do you mean:
This only affects functions that are executed remotely, <when or since> the secret value is injected to the function pod.

Reply:
Added a clarification.